manager/state/raft/raft.go

```diff
@@ -1498,9 +1505,11 @@ func (n *Node) registerNode(node *api.RaftMember) error {
 // ProposeValue calls Propose on the underlying raft library(etcd/raft) and waits
 // on the commit log action before returning a result
 func (n *Node) ProposeValue(ctx context.Context, storeAction []api.StoreAction, cb func()) error {
+	defer metrics.StartTimer(proposeLatencyTimer)()
```

Comment:
how does this defer call work?

Reply:
this isn't a defect in your code i just don't know how this works.

Reply:
StartTimer() starts the timer and returns a function which is deferred. The return function terminates the timer.